### PR TITLE
TCP-81: Add tests for post forms

### PR DESCRIPTION
**Pull request by daviddzhang**
Added tests for the various controller methods pertaining to post creation. Let me know if you think I need tests for the model (there aren't any validations on the model, so I didn't think there was anything worth testing) or view. Not sure how views are usually tested.

I also probably didn't need to, but I hooked up FactoryBot in the helper files so I could use create(:post) in these tests.

### CODE REVIEW

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  describe "GET #new" do
+    it "returns http success" do
+      get :new
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "POST #create" do
+    it "successfully creates a Post" do
+      post :create, params: { post: { title: "A title", text: "Some text" } }
```
Comment (daviddzhang) [inline spec/controllers/posts_controller_spec.rb:20]: Should I also be testing the route somewhere? As in should I be trying to make a POST to `/posts` at some point in these tests?

Comment (mshenton75) [reply spec/controllers/posts_controller_spec.rb:20]: I think this is sufficient 👍

File: spec/controllers/posts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      new_post = Post.last
+      expect(new_post.title).to eq("A title")
+      expect(new_post.text).to eq("Some text")
```
Comment (mshenton75) [inline spec/controllers/posts_controller_spec.rb:23]: I believe you could also use the following assertion:
```
expect(Post.where(title: 'A title', text: 'Some text')).to exist

**Review by mshenton75 — APPROVED** [review]
Nice, looks good to me!

As for testing views, we usually do this via integration tests.  We  can test for pages to have certain buttons, text, css selectors, etc. We also use Selenum Webdriver to simulate user actions, so we could test, for example, clicking on `create` actually commits a `Post` to the database.

**Pull request merged**